€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:1891]
Traceback (most recent call last):
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\Diana\pythonWorkspace\academy\app\classes\routes.py", line 231, in view_class
    return render_template(
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\templating.py", line 138, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\environment.py", line 930, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\environment.py", line 883, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\environment.py", line 857, in _load_template
    template = self.loader.load(self, name, globals)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\loaders.py", line 127, in load
    code = environment.compile(source, name, filename)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\environment.py", line 638, in compile
    self.handle_exception(source=source_hint)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\Diana\pythonWorkspace\academy\app\templates\class\view_class.html", line 269, in template
    {% endblock %}
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endblock'. You probably made a nesting mistake. Jinja is expecting this tag, but currently looking for 'elif' or 'else' or 'endif'. The innermost block that needs to be closed is 'if'.
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:1891]
Traceback (most recent call last):
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\Diana\pythonWorkspace\academy\app\models.py", line 211, in check_auth
    return func (*args, **kwargs)
  File "C:\Users\Diana\pythonWorkspace\academy\app\students\routes.py", line 441, in edit_student
    return render_template('students/edit_student.html', title="Edit Student", form=form, form2=form2, form3=form3, student=student, academy=academy)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\templating.py", line 137, in render_template
    return _render(
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask\templating.py", line 120, in _render
    rv = template.render(context)
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\environment.py", line 1090, in render
    self.environment.handle_exception()
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\Diana\pythonWorkspace\academy\app\templates\students\edit_student.html", line 3, in top-level template code
    {% from "_formhelpers.html" import render_field %}
  File "C:\Users\Diana\pythonWorkspace\academy\app\templates\base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 1, in top-level template code
    {% block doc -%}
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 4, in block "doc"
    {%- block html %}
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in block "html"
    {% block body -%}
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 23, in block "body"
    {% block content -%}
  File "C:\Users\Diana\pythonWorkspace\academy\app\templates\base.html", line 65, in block "content"
    {% block app_content %}{% endblock %}
  File "C:\Users\Diana\pythonWorkspace\academy\app\templates\students\edit_student.html", line 124, in block "app_content"
    {{ render_field(form2.lesson, id='lesson_name2', class='form-control lesson_', style="width:33%;") }}
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\runtime.py", line 679, in _invoke
    rv = self._func(*arguments)
  File "C:\Users\Diana\pythonWorkspace\academy\app\templates\_formhelpers.html", line 3, in template
    <dt> {{ field.label(class=field.id ) }}</dt>
  File "c:\users\diana\pythonworkspace\academy\venv\lib\site-packages\jinja2\environment.py", line 471, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'app.students.forms.AdditionalClassForm object' has no attribute 'lesson'
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
€(asctime)s €(levelname)s: €(message)s [in €pathnames:92]
